code/modules/projectiles/updated_projectiles/projectile.dm

```diff
@@ -249,14 +249,18 @@
 				while(++i <= 2 && hit_chance > 0) // This runs twice if necessary
 					hit_roll 					= rand(0, 99) //Our randomly generated roll
 					#if DEBUG_HIT_CHANCE
-					to_chat(world, "DEBUG: Hit Chance 1: [hit_chance], Hit Roll: [hit_roll]")
+					//to_chat(world, "DEBUG: Hit Chance 1: [hit_chance], Hit Roll: [hit_roll]")
 					#endif
-					if(hit_roll < 25 && !shot_from.sniper_target(A)) //Sniper targets more likely to hit
-						def_zone = pick(base_miss_chance)	// Still hit but now we might hit the wrong body part
-					if(!shot_from.sniper_target(A))
+					if(hit_roll < 25) //Sniper targets more likely to hit
+						if(shot_from && !shot_from.sniper_target(A)) //Avoid sentry run times
+							def_zone = pick(base_miss_chance)	// Still hit but now we might hit the wrong body part
+						else
+							def_zone = pick(base_miss_chance)	// Still hit but now we might hit the wrong body part
```

Comment:
This `else` is doing the same as the `if` check, making it irrelevant.
Check the whole logic here, please.